api/v1/MEL.php

```diff
@@ -410,86 +439,89 @@ public function GeometryExportName(string $name, int $layer_type = -1, bool $con
+     * @param PlanLayer[] $planLayers
+     *
      * @throws Exception
      */
     private function onGeometryExportNameResult(
         array $layer,
-        array $geometry,
+        array $planLayers,
         array &$result,
         ?int $geometryTypeFilter = null
     ): void {
         $result['debug-message'] ??= '';
         $result['debug-message'] .= 'hook activated for layer: '.$layer['layerId'].'.'.PHP_EOL;
-        $connManager = ConnectionManager::getInstance();
-        $conn = $connManager->getGameSessionEntityManager($this->getGameSessionId());
-        $qb = $conn->createQueryBuilder();
-        $qb
-            ->from('App:Policy', 'p')
-            ->select('pt.name, p.value as value, l.layerId')
-            ->innerJoin('p.type', 'pt')
-            ->leftJoin('p.policyLayers', 'pl')
-            ->leftJoin('pl.layer', 'l')
-            ->leftJoin('l.originalLayer', 'ol')
-            ->where('l.layerId = :layerId OR ol.layerId = :layerId')
-            ->setParameter('layerId', $layer['layerId']);
-        if ($geometryTypeFilter !== null) {
-            $qb
-                ->addSelect('pf.value as filterValue')
-                ->leftJoin('p.policyFilterLinks', 'pfl')
-                ->leftJoin('pfl.policyFilter', 'pf')
-                ->leftJoin('pf.type', 'pft')
-                ->andWhere('pft.name = :filterTypeName')
-                ->setParameter('filterTypeName', 'fleet')
-                ->andWhere('BIT_AND(pf.value,:filterValue) = :filterValue')
-                ->setParameter('filterValue', $geometryTypeFilter);
-        }
-        $policies = $qb
-            ->getQuery()
-            ->getResult(AbstractQuery::HYDRATE_ARRAY);
-        $result['debug-message'] .= 'policies: '.implode(',', array_column($policies, 'name')).'.'.PHP_EOL;
+
         // todo(MH) there is probably a way to get it working with
         //   ConnectionManager::getInstance()->getCachedServerManagerDbConnection()->getNativeConnection()
         $pdo = new \PDO("mysql:host=$_ENV[DATABASE_HOST];port=$_ENV[DATABASE_PORT]", 'root', '');
-        foreach ($policies as $policy) {
-            if ($policy['name'] == 'buffer') {
-                $this->applyBufferPolicy(
-                    $policy,
-                    collect($geometry)->filter(fn($g) => $g['layerId'] == $policy['layerId'])->all(),
-                    $result,
-                    $pdo
-                );
+        foreach ($planLayers as $planLayer) {
+            $planPolicies = $planLayer->getPlan()->getPlanPolicies()->toArray();
+            if (empty($planPolicies)) {
+                continue;
+            }
+            $result['debug-message'] .= 'policies: '.
+                implode(
+                    ',',
+                    array_map(fn(PlanPolicy $pp) => $pp->getPolicy()->getType()->getName(), $planPolicies)
+                ).'.'.PHP_EOL;
+            $planPolicies = $this->applyPolicyFilters($planPolicies, $geometryTypeFilter);
```

Comment:
apply filters like "schedule" and "fleet" that are applicable for different policy types